File: PayHeureUI/src/app/services/pointage.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Pointage, PointageCreateRequest } from '../models/Pointage';
import { API_BASE_URL } from '../config/api.config';

@Injectable({
  providedIn: 'root'
})
export class PointageService {

  private readonly apiUrl = `${API_BASE_URL}/pointages`;

  constructor(private http: HttpClient) {}

  /** Ajoute un badgeage pour le salarié désigné dans la requête. */
  creer(request: PointageCreateRequest): Observable<Pointage> {
    return this.http.post<Pointage>(this.apiUrl, request);
  }
}

File: src/main/java/com/example/payheurebackend/service/PointageService.java
package com.example.payheurebackend.service;

import com.example.payheurebackend.dto.PointageCreateRequest;
import com.example.payheurebackend.dto.PointageResponse;

/** Enregistrement des pointages (badgeages bruts) d'un salarié. */
public interface PointageService {

    /**
     * Enregistre un nouveau badgeage pour le salarié désigné dans la requête.
     *
     * @throws com.example.payheurebackend.exception.ResourceNotFoundException si le salarié n'existe pas
     */
    PointageResponse creer(PointageCreateRequest request);
}
